numba_dpex/core/kernel_interface/dispatcher.py

```diff
@@ -0,0 +1,460 @@
+# SPDX-FileCopyrightText: 2022 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+from inspect import signature
+from warnings import warn
+
+import dpctl
+import dpctl.program as dpctl_prog
+from numba.core.types import Array as ArrayType
+
+from numba_dpex import config
+from numba_dpex.core.descriptor import dpex_target
+from numba_dpex.core.exceptions import (
+    ComputeFollowsDataInferenceError,
+    ExecutionQueueInferenceError,
+    IllegalRangeValueError,
+    InvalidKernelLaunchArgsError,
+    SUAIProtocolError,
+    UnknownGlobalRangeError,
+    UnsupportedBackendError,
+    UnsupportedNumberOfRangeDimsError,
+    UnsupportedWorkItemSizeError,
+)
+from numba_dpex.core.kernel_interface.arg_pack_unpacker import Packer
+from numba_dpex.core.kernel_interface.spirv_kernel import SpirvKernel
+from numba_dpex.dpctl_iface import USMNdArrayType
+
+
+def get_ordered_arg_access_types(pyfunc, access_types):
+    # Construct a list of access type of each arg according to their position
+    ordered_arg_access_types = []
+    sig = signature(pyfunc, follow_wrapped=False)
+    for idx, arg_name in enumerate(sig.parameters):
+        if access_types:
+            for key in access_types:
+                if arg_name in access_types[key]:
+                    ordered_arg_access_types.append(key)
+        if len(ordered_arg_access_types) <= idx:
+            ordered_arg_access_types.append(None)
+
+    return ordered_arg_access_types
+
+
+class Dispatcher(object):
+    """Creates a Kernel object from a @kernel decorated function and enqueues
+    the Kernel object on a specified device.
+    """
+
+    # The list of SYCL backends supported by the Dispatcher
+    _supported_backends = ["opencl", "level_zero"]
+
+    def __init__(
+        self,
+        pyfunc,
+        debug_flags=None,
+        compile_flags=None,
+        array_access_specifiers=None,
+    ):
+        self.typingctx = dpex_target.typing_context
```

Comment:
`dpex_target` is a global object coming from an external module, why do we always keep it in some local variable? Is there any specific reason for this?